lib/source/google_key_source.ex

```diff
@@ -1,18 +1,30 @@
 defmodule ExFirebaseAuth.KeySource.Google do
   @moduledoc false
 
-  @endpoint_url "https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com"
+  @endpoint_urls [
+    "https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com",
+    "https://www.googleapis.com/identitytoolkit/v3/relyingparty/publicKeys"
+  ]
 
   @behaviour ExFirebaseAuth.KeySource
 
   def fetch_certificates do
-    with {:ok, %Finch.Response{body: body}} <-
-           Finch.build(:get, @endpoint_url) |> Finch.request(ExFirebaseAuthFinch),
-         {:ok, json_data} <- Jason.decode(body) do
-      {:ok, convert_to_jose_keys(json_data)}
+    results =
+      @endpoint_urls
+      |> Enum.map(fn endpoint_url ->
+        with {:ok, %Finch.Response{body: body}} <-
+               Finch.build(:get, endpoint_url) |> Finch.request(ExFirebaseAuthFinch),
+             {:ok, json_data} <- Jason.decode(body) do
+          {:ok, convert_to_jose_keys(json_data)}
+        else
+          _ -> :error
+        end
+      end)
+
+    if Enum.any?(results, &(&1 == :error)) do
+      :error
     else
```

Comment:
I'm not sure if we should completely fail fetching certificates if only one of them failed? I think it would be better to return at least the certificates that we could fetch, and log an error.

Reply:
Ah gotcha, I was not aware of that dependency. Nevermind then.